mmdet/datasets/pipelines/transforms.py

```diff
@@ -400,7 +400,7 @@ def __call__(self, results):
 
             # filter and crop the masks
             if 'gt_masks' in results:
-                results['gt_masks'] = results['gt_masks'].crop(
+                results['gt_masks'] = results['gt_masks'][valid_inds].crop(
```

Comment:
If masks here are in format Polygon then there will raise an error. Currently polygon doesn't support mask inds.
You can enable polygon mask by setting `poly2mask=False` in train_pipeline. 
```
train_pipeline = [
    dict(
        type='LoadAnnotations',
        with_bbox=True,
        with_mask=True,
        poly2mask=False),
```
We need to convert `valid_inds` as following:
`valid_inds = valid_inds.nonzero()[0]`